templates/web/base/admin/report-category.html

```diff
@@ -7,7 +7,7 @@
   [% IF categories.size %]
     <optgroup label="[% loc('Available categories') %]">
       [% FOREACH cat IN categories %]
-        <option[% ' selected' IF problem.category == cat %]>[% cat | html %]</option>
+        <option[% ' selected' IF problem.category == cat %] data-defect-types='[% category_defect_types.$cat %]'>[% cat | html %]</option>
```

Comment:
This doesn't feel like the right place for this, because this template is also included by the main admin's report edit page. Would it not fit better inside the loop at https://github.com/mysociety/fixmystreet/blob/a3c7641247b16fd4fd959eb95c54f1f81c6f4ba7/templates/web/base/report/_inspect.html#L42 to match the existing priorities code that appears to be doing the same functionality for the priority dropdown? (Would also therefore be good to match (share a function) how that does it for consistency, ie. it is deleting all but the first option, not deleting them all and then putting back a `-` option; could/should the priorities code use JSON too?)